src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -343,35 +372,25 @@ impl<'a, 'tcx> GatherLoanCtxt<'a, 'tcx> {
         // Create the loan record (if needed).
         let loan = match restr {
             RestrictionResult::Safe => {
-                // No restrictions---no loan record necessary
+                let loan_scope = loan_scope_from_region(loan_region);
+                debug!("loan_scope = {:?}", loan_scope);
+
+                let borrow_scope = region::CodeExtent::Misc(borrow_id);
+                let loan_scope = self.compute_gen_scope(borrow_scope, loan_scope);
```

Comment:
Would this be enough to show the span when the loan is being held (this is for temporary loans, I believe)? It seems correct when visualizing, but I'm definitely no expert on the internals of the borrow checker.